src/service-worker.test.js

```diff
@@ -29,7 +27,11 @@ beforeEach(() => {
       shownNotifications.push({ title, options }),
   };
   global.clients = {
-    openWindow: url => openedWindows.push(url),
+    openWindow: url => {
+      const client = new FakeWindowClient({ url });
+      clients.push(client);
+      return Promise.resolve(client);
+    },
     matchAll: () => Promise.resolve(clients),
```

Comment:
To test this, I had to rework the mock `Clients.openWindow()` function to be closer to what the actual [Clients.openWindow()](https://developer.mozilla.org/en-US/docs/Web/API/Clients/openWindow) would do - create a new [WindowClient](https://developer.mozilla.org/en-US/docs/Web/API/WindowClient), that will be returned by [Clients.matchAll()](https://developer.mozilla.org/en-US/docs/Web/API/Clients/matchAll)